src/main/java/domain/LottoNumber.java

```diff
@@ -0,0 +1,61 @@
+package domain;
+
+import java.util.Objects;
+import java.util.stream.IntStream;
+
+public class LottoNumber implements Comparable<LottoNumber> {
+
+    private static final String ERROR_MESSAGE_NOT_IN_RANGE = "유효한 로또 번호가 아닙니다.";
+
+    private static final int MINIMUM_VALUE = 1;
+    private static final int MAXIMUM_VALUE = 45;
+
+    private static final LottoNumber[] LOTTO_NUMBERS = new LottoNumber[MAXIMUM_VALUE + 1];
+
+    static {
+        IntStream.rangeClosed(MINIMUM_VALUE, MAXIMUM_VALUE)
+            .forEach(number -> LOTTO_NUMBERS[number] = new LottoNumber(number));
+    }
+
+    private final int value;
+
+    private LottoNumber(int value) {
+        this.value = value;
+    }
+
+    public static LottoNumber valueOf(final int number) {
+        validateInRange(number);
+        return LOTTO_NUMBERS[number];
+    }
+
+    private static void validateInRange(int value) {
+        if (value < MINIMUM_VALUE || value > MAXIMUM_VALUE) {
+            throw new IllegalArgumentException(ERROR_MESSAGE_NOT_IN_RANGE);
+        }
+    }
+
+    public int getValue() {
+        return value;
+    }
+
+    @Override
+    public int compareTo(LottoNumber other) {
+        return this.value - other.value;
+    }
+
```

Comment:
```suggestion
    @Override
    public int compareTo(LottoNumber other) {
        return Integer.compare(this.value, other.value);
    }
```
Integer에서 제공하는 메서드를 사용하는 방법도 있답니다 :)